yt/yt/server/node/data_node/master_connector.cpp

```diff
@@ -334,9 +334,42 @@ class TMasterConnector
         YT_VERIFY(delta->State == EMasterConnectorState::Registered);
 
         delta->State = EMasterConnectorState::Online;
-        YT_VERIFY(delta->AddedSinceLastSuccess.empty());
-        YT_VERIFY(delta->RemovedSinceLastSuccess.empty());
-        YT_VERIFY(delta->ChangedMediumSinceLastSuccess.empty());
+
+        {
+            YT_VERIFY(delta->AddedSinceLastSuccess.empty());
+            YT_VERIFY(delta->RemovedSinceLastSuccess.empty());
+            YT_VERIFY(delta->ChangedMediumSinceLastSuccess.empty());
+
+            delta->AddedSinceLastSuccess = delta->AddedSinceLastFullHeartbeatTry;
+            delta->RemovedSinceLastSuccess = delta->RemovedSinceLastFullHeartbeatTry;
+            delta->ChangedMediumSinceLastSuccess = delta->ChangedMediumSinceLastFullHeartbeatTry;
+
+            if (!delta->AddedSinceLastSuccess.empty() ||
+                !delta->RemovedSinceLastSuccess.empty() ||
+                !delta->ChangedMediumSinceLastSuccess.empty())
+            {
+                YT_LOG_INFO(
```

Comment:
Won't it be too many log messages in case of location disable?

Reply:
It shouldn't, it's only when sending full heartbeats. Maximum number of chunk IDs in the log is 3&